### DISCUSSION TITLE
`--mermaid` filter slow?

### DISCUSSION THREAD

**Discussion by reagle** [jgm/pandoc#10081] [Q&A] [answered]
I appreciate being able to included mermaid diagrams in my markdown, but it takes about 2 seconds per diagram per conversion. It's not the end of the world, but annoyingly slow once I have a few diagrams. Is this typical? For example, it takes over 10s to generate [this html](https://reagle.org/joseph/2016/min-eg/test-diagram.html) from [this markdown](https://reagle.org/joseph/2016/min-eg/test-diagram.md).

```
❯ pandoc --version
pandoc 3.3
Features: +server +lua
Scripting engine: Lua 5.4
User data directory: /Users/reagle/.pandoc
❯ macv
macOS 14.5 23F79 arm64
```

**Comment by reagle** [comment]
BTW:  Anyone ever had [SVG files working with odt output format](https://github.com/raghur/mermaid-filter/issues/132)?

**Comment by tarleb** [comment, marked as answer]
Yes, using mermaid this way is slow. The reason is that the filter has to fire up a whole browser each time it converts a diagram, and that takes its time. I don't know which exact tool you're using, but I get very similar times with the `diagram.lua` filter. Enabling caching for that filter makes subsequent runs a lot faster, maybe your filter of choice has a similar option?

**Comment by reagle** [comment]
Hi Albert, I use https://github.com/raghur/mermaid-filter . It doesn't look to support a cache option, perhaps I should give diagram.lua a try!
